pkg/kubelet/volumes_linux.go

```diff
@@ -0,0 +1,63 @@
+// +build linux
+
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+// Bitmask to OR with current ownership of volumes that allow ownership management by the Kubelet
+const managedOwnershipBitmask = os.FileMode(0660)
+
+// manageVolumeOwnership modifies the given volume to be owned by fsGroup.
+func (kl *Kubelet) manageVolumeOwnership(pod *api.Pod, volSpec *volume.Spec, builder volume.Builder, fsGroup int64) error {
+	return filepath.Walk(builder.GetPath(), func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// TODO: This is a very brittle and kludgey way to avoid having to re-stat the path
```

Comment:
I think this comment is wrong top-to-bottom.  Golang structs don't want to carry OS-specific stuff (frankly I am surprised Mode is there, but it's at least abstract).  This code is already in a *_linux.go file.  FileInfo.Sys is provided EXACTLY so that code can get at OS-specifics if needed.